### PR TITLE
Add init with `PartialSyntaxNodeString` on `SwitchStmtSyntax`

### PR DESCRIPTION
**Pull request by kimdv**

### CODE REVIEW

File: Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift [hunk 1]
```diff
@@ -105,10 +105,21 @@ public extension IfStmtSyntax {
     self.elseKeyword = elseBody != nil ? .keyword(.else) : nil
   }
 
-  init(_ signature: String, @CodeBlockItemListBuilder bodyBuilder: () -> CodeBlockItemListSyntax, elseIf: IfStmtSyntax) {
+  init(_ signature: PartialSyntaxNodeString, @CodeBlockItemListBuilder bodyBuilder: () -> CodeBlockItemListSyntax, elseIf: IfStmtSyntax) {
```
Comment (kimdv) [inline Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift:108]: Also fixed a warning

**Comment by kimdv** [comment]
@swift-ci please test

**Review by ahoppen — APPROVED** [review]
Good catch that `IfStmtSyntax` still took a `String`. And the `SwitchStmtSyntax` initializer also looks useful. 

We probably also want one for `VariableDecl` to create computed properties (they are currently so verbose in CodeGenerations)

**Comment by kimdv** [comment]
@swift-ci please test

**Pull request merged**